### PR TITLE
fix: tenantid in config functions

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-  https://github.com/supertokens/supertokens-node/pull/471

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/recipe/openid/types.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export type RecipeInterface = {
         payload?: any;
         validitySeconds?: number;
         useStaticSigningKey?: boolean;
+        tenantId: string;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/openid/types.ts:103]: this is not needed

File: lib/ts/recipe/passwordless/types.ts [hunk 2]
```diff
@@ -59,12 +59,13 @@ export type TypeInput = (
                   // Unlikely, but someone could display this (or a derived thing) to identify the device
                   preAuthSessionId: string;
               },
+              tenantId: string,
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/types.ts:62]: remove the deprecated createAndSendCustomTextMessage and other functions from other recipes

File: lib/ts/recipe/session/types.ts [hunk 3]
```diff
@@ -182,6 +184,7 @@ export interface VerifySessionOptions {
     overrideGlobalClaimValidators?: (
         globalClaimValidators: SessionClaimValidator[],
         session: SessionContainerInterface,
+        tenantId: string,
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/types.ts:187]: this is not needed

File: lib/ts/recipe/thirdparty/types.ts [hunk 4]
```diff
@@ -64,10 +64,11 @@ type CommonProviderConfig = {
     validateIdTokenPayload?: (input: {
         idTokenPayload: { [key: string]: any };
         clientConfig: ProviderConfigForClientType;
+        tenantId: string;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdparty/types.ts:67]: don't need it

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/emailpassword/utils.ts [hunk 5]
```diff
@@ -60,12 +60,7 @@ export function validateAndNormaliseUserInput(
          * createAndSendCustomEmail implementation which calls our supertokens API
          */
         if (emailService === undefined) {
-            emailService = new BackwardCompatibilityService(
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailpassword/utils.ts:63]: what about changing the types of BackwardCompatibilityService?

File: lib/ts/recipe/emailverification/utils.ts [hunk 6]
```diff
@@ -39,11 +39,7 @@ export function validateAndNormaliseUserInput(
          * createAndSendCustomEmail implementation which calls our supertokens API
          */
         if (emailService === undefined) {
-            emailService = new BackwardCompatibilityService(
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/utils.ts:42]: change type of this?

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts [hunk 7]
```diff
@@ -29,7 +30,8 @@ function defaultCreateAndSendCustomEmail(appInfo: NormalisedAppinfo) {
             urlWithLinkCode?: string;
             codeLifetime: number;
         },
-        _: any
+        _: string,
+        __: any
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:34]: ??

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts [hunk 8]
```diff
@@ -105,30 +107,12 @@ export default class BackwardCompatibilityService
             // Unlikely, but someone could display this (or a derived thing) to identify the device
             preAuthSessionId: string;
         },
+        tenantId: string,
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:110]: ??

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts [hunk 9]
```diff
@@ -140,6 +124,7 @@ export default class BackwardCompatibilityService
                 preAuthSessionId: input.preAuthSessionId,
                 codeLifetime: input.codeLifetime,
             },
+            input.tenantId === undefined ? DEFAULT_TENANT_ID : input.tenantId,
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:127]: ??

File: lib/ts/recipe/passwordless/smsdelivery/services/backwardCompatibility/index.ts [hunk 10]
```diff
@@ -31,7 +32,8 @@ function defaultCreateAndSendCustomSms(_: NormalisedAppinfo) {
             urlWithLinkCode?: string;
             codeLifetime: number;
         },
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/smsdelivery/services/backwardCompatibility/index.ts:34]: revisit this

File: lib/ts/recipe/thirdpartyemailpassword/utils.ts [hunk 11]
```diff
@@ -53,12 +51,7 @@ export function validateAndNormaliseUserInput(
          * createAndSendCustomEmail implementation
          */
         if (emailService === undefined) {
-            emailService = new BackwardCompatibilityService(
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartyemailpassword/utils.ts:56]: change this type

File: lib/ts/recipe/thirdpartypasswordless/types.ts [hunk 12]
```diff
@@ -63,7 +63,10 @@ export type User = (
...
     | {
           contactMethod: "PHONE";
-          validatePhoneNumber?: (phoneNumber: string) => Promise<string | undefined> | string | undefined;
+          validatePhoneNumber?: (
+              phoneNumber: string,
+              tenantId: string
+          ) => Promise<string | undefined> | string | undefined;
 
           // Override to use custom template/contact method
           /**
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartypasswordless/types.ts:72]: createAndSendCustomTextMessage needs to be removed

Comment (rishabhpoddar) [reply lib/ts/recipe/thirdpartypasswordless/types.ts:72]: Even the email version

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/openid/types.ts (hunk 1 above)
Comment (sattvikc) [reply lib/ts/recipe/openid/types.ts]: Done

File: lib/ts/recipe/passwordless/types.ts (hunk 2 above)
Comment (sattvikc) [reply lib/ts/recipe/passwordless/types.ts]: Done

File: lib/ts/recipe/session/types.ts (hunk 3 above)
Comment (sattvikc) [reply lib/ts/recipe/session/types.ts]: Done

File: lib/ts/recipe/thirdparty/types.ts (hunk 4 above)
Comment (sattvikc) [reply lib/ts/recipe/thirdparty/types.ts]: Done

File: lib/ts/recipe/emailpassword/utils.ts (hunk 5 above)
Comment (sattvikc) [reply lib/ts/recipe/emailpassword/utils.ts:63]: Done

File: lib/ts/recipe/emailverification/utils.ts (hunk 6 above)
Comment (sattvikc) [reply lib/ts/recipe/emailverification/utils.ts:42]: Done

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts (hunk 7 above)
Comment (sattvikc) [reply lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:34]: Done

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts (hunk 8 above)
Comment (sattvikc) [reply lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:110]: Done

File: lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts (hunk 9 above)
Comment (sattvikc) [reply lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts:127]: Done

File: lib/ts/recipe/passwordless/smsdelivery/services/backwardCompatibility/index.ts (hunk 10 above)
Comment (sattvikc) [reply lib/ts/recipe/passwordless/smsdelivery/services/backwardCompatibility/index.ts:34]: Done

File: lib/ts/recipe/thirdpartyemailpassword/utils.ts (hunk 11 above)
Comment (sattvikc) [reply lib/ts/recipe/thirdpartyemailpassword/utils.ts:56]: Done

File: lib/ts/recipe/thirdpartypasswordless/types.ts (hunk 12 above)
Comment (sattvikc) [reply lib/ts/recipe/thirdpartypasswordless/types.ts:72]: Done

File: lib/ts/recipe/emailpassword/utils.ts [hunk 13]
```diff
@@ -54,10 +54,8 @@ export function validateAndNormaliseUserInput(
         let emailService = config?.emailDelivery?.service;
         /**
          * following code is for backward compatibility.
-         * if user has not passed emailDelivery config, we
-         * use the createAndSendCustomEmail config. If the user
-         * has not passed even that config, we use the default
-         * createAndSendCustomEmail implementation which calls our supertokens API
+         * if user has not passed emailDelivery config, we use the default
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailpassword/utils.ts:57]:
```suggestion
         * if user has not passed emailService config, we use the default
```

File: lib/ts/recipe/emailverification/utils.ts [hunk 14]
```diff
@@ -33,10 +33,8 @@ export function validateAndNormaliseUserInput(
         let emailService = config.emailDelivery?.service;
         /**
          * following code is for backward compatibility.
-         * if user has not passed emailDelivery config, we
-         * use the createAndSendCustomEmail config. If the user
-         * has not passed even that config, we use the default
-         * createAndSendCustomEmail implementation which calls our supertokens API
+         * if user has not passed emailDelivery config, we use the default
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/utils.ts:36]:
```suggestion
         * if user has not passed emailService config, we use the default
```

File: lib/ts/recipe/passwordless/utils.ts [hunk 15]
```diff
@@ -48,10 +48,8 @@ export function validateAndNormaliseUserInput(
 
         /**
          * following code is for backward compatibility.
-         * if user has not passed emailDelivery config, we
-         * use the createAndSendCustomEmail config. If the user
-         * has not passed even that config, we use the default
-         * createAndSendCustomEmail implementation
+         * if user has not passed emailDelivery config, we use the default
```
Comment (rishabhpoddar) [inline lib/ts/recipe/passwordless/utils.ts:51]:
```suggestion
         * if user has not passed emailService config, we use the default
```

File: lib/ts/recipe/thirdpartyemailpassword/utils.ts [hunk 16]
```diff
@@ -45,10 +45,8 @@ export function validateAndNormaliseUserInput(
         let emailService = config?.emailDelivery?.service;
         /**
          * following code is for backward compatibility.
-         * if user has not passed emailDelivery config, we
-         * use the createAndSendCustomEmail config. If the user
-         * has not passed even that config, we use the default
-         * createAndSendCustomEmail implementation
+         * if user has not passed emailDelivery config, we use the default
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartyemailpassword/utils.ts:48]:
```suggestion
         * if user has not passed emailService config, we use the default
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Pull request merged**